### PR TITLE
datasets.make_regression as Dask dataframe and series

### PR DESCRIPTION
**Pull request by mmccarty**
- Added `return_dataframes` wrapper to repackage Xs and ys
- Create `make_regression_df` using wrapper
- Refactored `make_classification_df` to reuse wrapper

### CODE REVIEW

**Comment by mmccarty** [comment]
cc @dhavide

**Comment by mmccarty** [comment]
cc @TomAugspurger I promised this a while back. Do you think I should do anything fancy for the docstrings? This just specifies the new return values.

File: dask_ml/datasets.py [hunk 1]
```diff
@@ -388,75 +389,50 @@ def random_date(start, end):
...
-                    columns=["date"],
-                ),
-            ],
-            axis=1,
-        )
 
-    return X_df, y_series
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        """
```
Comment (TomAugspurger) [inline dask_ml/datasets.py:399]: Is this the eventual docstring for `make_classification_df`? If so, can you add the additional parameters to the docstring (dates, response_rate, etc.)

Comment (mmccarty) [reply dask_ml/datasets.py:399]: It is. Yes, I'll do that and add a note to see the array versions.

**Pull request closed without merging**